Added better error handling in LogglySink

danjagnow opened the pull request:
Updated the `EmitBatchAsync` method in `LogglySink` to handle potential exceptions and evaluate the `LogResponse` when calling the `Log` method of the `LogglyClient`.  It now writes to `SelfLog` when problems occur to enable easier debugging.

I believe this may help with the problems described in #39.

Also removed a digit from the `PackageReference` for `Serilog.Sinks.RollingFileAlternate` in **sampleDurableLogger.csproj** to get local builds working.

danjagnow committed:
Build fix for sampleDurableLogger.csproj

Removed a digit from the PackageReference for Serilog.Sinks.RollingFileAlternate in sampleDurableLogger.csproj to get local builds working.

danjagnow committed:
Added better error handling in LogglySink

Updated the EmitBatchAsync method in LogglySink to handle potential exceptions and evaluate the LogResponse when calling the Log method of the LogglyClient.  It now writes to SelfLog when problems occur to enable easier debugging.

danjagnow commented:
@nblumhardt or @MiguelAlho, just wondering if you've had a chance to look at this PR.  I know life gets busy.  😄

nblumhardt commented on src/Serilog.Sinks.Loggly/Sinks/Loggly/LogglySink.cs:
```diff
@@ -80,7 +81,24 @@ public LogglySink(IFormatProvider formatProvider, int batchSizeLimit, TimeSpan p
+                        break;
+                }
+            }
+            catch (Exception ex)
```
The exception shouldn't be caught here, but rather propagate up to Serilog (just removing the try/catch block here is fine).

danjagnow replied:
Done.  Thanks for taking a look!

nblumhardt reviewed (changes requested):
Thanks for the nudge! The response code logging looks like a great improvement. Exception handling in sinks is discouraged because exceptions are (should be!) handled centrally and consistently by Serilog.

danjagnow committed:
Removed try/catch block in LogglySink

Removed the try/catch block in the EmitBatchAsync method of LogglySink to allow Serilog to handle exceptions centrally, based on PR feedback.

nblumhardt commented:
Thanks! 👍
